server/post_test.go

```diff
@@ -0,0 +1,340 @@
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(&model.Channel{Id: "testChannelID", Type: model.ChannelTypeDirect}, nil)
+				api.On("GetChannelMembers", "testChannelID", 0, 100).Return(model.ChannelMembers{}, nil)
+				api.On("CreatePost", mock.Anything).Return(&model.Post{}, nil)
+				testJoinURL := "testJoinURL"
+				client.On("CreateMeeting").Return(&msgraph.OnlineMeeting{JoinURL: &testJoinURL}, nil)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			api.ExpectedCalls = nil
+			client.ExpectedCalls = nil
+
+			tt.setup()
+
+			_, _, err := p.postMeeting(tt.creator, tt.channelID, tt.topic)
+
+			if tt.expectedError != "" {
+				require.Error(t, err)
+				require.Equal(t, tt.expectedError, err.Error())
+			} else {
+				require.NoError(t, err)
+			}
+
+			api.AssertExpectations(t)
+		})
+	}
+}
+
+func TestPostConfirmCreateOrJoin(t *testing.T) {
+	testCases := []struct {
+		name            string
+		provider        string
+		expectedMessage string
+	}{
+		{
+			name:            "Provider is msteams",
+			provider:        "msteams",
+			expectedMessage: "There is another recent meeting created on this channel.",
+		},
+		{
+			name:            "Provider is dummyProvider",
+			provider:        "dummyProvider",
+			expectedMessage: "There is another recent meeting created on this channel with dummyProvider.",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			channelID := "test_channel_id"
+			meetingURL := "https://example.com/meeting"
+			topic := "Test Meeting"
+			userID := "test_user_id"
+			creatorName := "test_creator"
+
+			api := &plugintest.API{}
+			client := &MockClient{}
+			p := &Plugin{
+				MattermostPlugin: plugin.MattermostPlugin{
+					API: api,
+				},
+				client:    client,
+				botUserID: "testBotUserID",
+			}
+			expectedPost := &model.Post{
+				UserId:    p.botUserID,
+				ChannelId: channelID,
+				Message:   tt.expectedMessage,
+				Type:      "custom_mstmeetings",
+				Props: map[string]interface{}{
+					"type":                     "custom_mstmeetings",
+					"meeting_link":             meetingURL,
+					"meeting_status":           postTypeConfirm,
+					"meeting_personal":         true,
+					"meeting_topic":            topic,
+					"meeting_creator_username": creatorName,
+					"meeting_provider":         tt.provider,
+				},
+			}
+			api.On("SendEphemeralPost", userID, mock.AnythingOfType("*model.Post")).Return(expectedPost, nil)
+
+			post := p.postConfirmCreateOrJoin(meetingURL, channelID, topic, userID, creatorName, tt.provider)
+			require.Equal(t, tt.expectedMessage, post.Message)
+		})
+	}
+}
+
+func TestPostConnect(t *testing.T) {
+	api := &plugintest.API{}
+	p := &Plugin{
+		MattermostPlugin: plugin.MattermostPlugin{
+			API: api,
+		},
+		botUserID: "testBostUserID",
+	}
```

Comment:
Should not be on global level, this can lead to API being using mocks from other testcases

Reply:
Discussed